doc/user/content/sql/types/map.md

```diff
@@ -0,0 +1,206 @@
+---
+title: "map Data Type"
+description: "Expresses a map"
+menu:
+  main:
+    parent: 'sql-types'
+aliases:
+  - /sql/types/map
+---
+
+{{< version-added v0.5.3 >}}
+
+`map` data expresses an unordered map with [`text`](../text) keys and an
+arbitrary uniform value type.
+
+Detail | Info
+-------|------
+**Quick Syntax** | `'{a=>123.4, b=>111.1}'::map[text=>double]'`
+**Size** | Variable
+
+## Syntax
+
+{{< diagram "type-map.svg" >}}
+
+Field | Use
+------|-----
+_map&lowbar;string_ | A well-formed map object.
+_value&lowbar;type_ | The [type](../../types) of the map's values.
+
+## Map operators
+
+Materialize supports the following map operators.
+
+### Operators
+
+{{% map-operators %}}
+
+## Details
+
+### Construction
+
+A well-formed `map` is a collection of `key => value` mappings separated
+by commas. Each individual `map` must be correctly contained by a set of
+curly braces (`{}`).
+
+You can construct maps from strings using the following syntax:
+```sql
+SELECT '{a=>123.4, b=>111.1}'::map[text=>double] as m;
+```
+```nofmt
+  m
+------------------
+ {a=>123.4,b=>111.1}
+```
+
+
+You can create nested maps the same way:
+```sql
+SELECT '{a=>{b=>{c=>d}}}'::map[text=>map[text=>map[text=>text]]] as nested_map;
+```
+```nofmt
+  nested_map
+------------------
+ {a=>{b=>{c=>d}}}
+```
+
+### Constraints
+
+- `map` keys must be of type [`text`](../text).
+- `map` values can be of any [type](../../types) as long as the type is uniform.
+- `map` keys must be unique. If duplicate keys are present in a map, only one of the
+  (`key`, `value`) pairs will be retained. There is no guarantee which will be retained.
+
+## Examples
+
+### Operators
+
+#### Field access as `map` (`->`)
```

Comment:
I don't think "Field access" is the right term here; seems like maybe this is "value access" or "Retrieve value with key"?